lib/Controller/ConfigController.php

```diff
@@ -433,4 +445,106 @@ public function checkConfig(): DataResponse {
 			'authorization_header' => $_SERVER['HTTP_AUTHORIZATION'],
 		]);
 	}
+
+	/**
+	 * @NoCSRFRequired
+	 * set up integration
+	 * @param array<string, string> $values
+	 *
+	 * @return DataResponse
+	 *
+	 */
+	public function setUpIntegration(array $values): DataResponse {
+		try {
+			// for POST all the keys must be provided so keyType = mustHaveKeys
+			OpenProjectAPIService::validateIntegrationSetupInformation($values, 'mustHaveKeys');
+			$status = $this->setUpIntegrationConfig($values);
+			$result = $this->recreateOauthClientInformation();
+			if ($status['oPOAuthTokenRevokeStatus'] !== '') {
+				$result['openproject_revocation_status'] = $status['oPOAuthTokenRevokeStatus'];
+			}
+			return new DataResponse($result);
+		} catch (\Exception $e) {
+			return new DataResponse([
+				"error" => $e->getMessage()
+			], Http::STATUS_BAD_REQUEST);
+		}
+	}
+
+	/**
+	 * @NoCSRFRequired
+	 *
+	 * update integration
+	 *
+	 * @param array<string, string> $values
+	 *
+	 *
+	 * @return DataResponse
+	 */
+	public function updateIntegration(array $values): DataResponse {
+		try {
+			// for PUT key information can be optional so keyType = allowedKeys
+			OpenProjectAPIService::validateIntegrationSetupInformation($values, 'allowedKeys');
+			$status = $this->setUpIntegrationConfig($values);
+			$oauthClientInternalId = $this->config->getAppValue(Application::APP_ID, 'nc_oauth_client_id', '');
+			if ($oauthClientInternalId !== '') {
+				$id = (int)$oauthClientInternalId;
+				$result = $this->oauthService->getClientInfo($id);
+				if ($status['oPOAuthTokenRevokeStatus'] !== '') {
+					$result['openproject_revocation_status'] = $status['oPOAuthTokenRevokeStatus'];
+				}
+				return new DataResponse($result);
+			}
+			return new DataResponse([
+				"error" => 'could not find nextcloud oauth client for openproject'
+			], Http::STATUS_INTERNAL_SERVER_ERROR);
+		} catch (\Exception $e) {
+			return new DataResponse([
+				"error" => $e->getMessage()
+			], Http::STATUS_BAD_REQUEST);
+		}
+	}
+
+	/**
+	 * @NoCSRFRequired
+	 *
+	 * reset integration
+	 *
+	 *
+	 * @return DataResponse
+	 */
+	public function resetIntegration(): DataResponse {
+		$values = [
+			'openproject_instance_url' => null,
+			'openproject_client_id' => null,
+			'openproject_client_secret' => null,
+			'default_enable_navigation' => null,
+			'default_enable_unified_search' => null
+		];
+		try {
+			$status = $this->setUpIntegrationConfig($values);
+			$result = ["status" => true];
+			if ($status['oPOAuthTokenRevokeStatus'] !== '') {
+				$result['openproject_revocation_status'] = $status['oPOAuthTokenRevokeStatus'];
+			}
+			return new DataResponse($result);
+		} catch (\Exception $e) {
+			return new DataResponse([
+				'error' => $e->getMessage()
+			], Http::STATUS_INTERNAL_SERVER_ERROR);
+		}
+	}
+
+
+	/**
+	 * @return array<mixed>
+	 */
+	public function recreateOauthClientInformation(): array {
```

Comment:
this is only used in the own class
```suggestion
	private function recreateOauthClientInformation(): array {
```